### PR TITLE
[IR Runtime] Support Max, Min, ToSet, CountDistinct in Accumulator

### PR DESCRIPTION
**Pull request by BingqingLyu**
## What do these changes do?

1. Support Max, Min, ToSet, CountDistinct in Accumulator. i.e., we support fold/group with these accum types.
2. Add uts for accumulators of Max, Min, ToSet, CountDistinct.

## Related issue number

Fixes

### CODE REVIEW

File: research/query_service/ir/runtime/src/process/operator/accum/accumulator.rs [hunk 1]
```diff
@@ -110,3 +114,236 @@ impl<D: Decode> Decode for ToList<D> {
...
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "max={:?}", self.max)
+    }
+}
+
+unsafe impl<D: Send> Send for Maximum<D> {}
+
+impl<D: Debug + Send + PartialOrd + 'static> Accumulator<D, Option<D>> for Maximum<D> {
+    fn accum(&mut self, next: D) -> FnExecResult<()> {
+        if let Some(pre) = self.max.take() {
```
Comment (longbinlai) [inline research/query_service/ir/runtime/src/process/operator/accum/accumulator.rs:182]: For both Maximum and Minimum, it is unnecessary to take the element out of the Option and then reassign, try:
```
if let Some(pre) = self.max.as_mut() {
    if pre < next {
        *pre = next;
    }
} else {
    self.max = Some(next);
}
```

File: research/query_service/ir/runtime/src/process/operator/accum/accumulator.rs [hunk 2]
```diff
@@ -110,3 +114,236 @@ impl<D: Decode> Decode for ToList<D> {
...
+        let mut result = Vec::with_capacity(self.inner.len());
+        let set = std::mem::replace(&mut self.inner, HashSet::new());
+        result.extend(set.into_iter());
+        Ok(result)
+    }
+}
+
+impl<D: Encode + Eq + Hash> Encode for ToSet<D> {
+    fn write_to<W: WriteExt>(&self, writer: &mut W) -> io::Result<()> {
+        writer.write_u32(self.inner.len() as u32)?;
```
[the remark this answers is not in the document]
Comment (BingqingLyu) [reply research/query_service/ir/runtime/src/process/operator/accum/accumulator.rs]: Done

File: research/query_service/ir/runtime/src/process/operator/accum/accumulator.rs [hunk 3]
```diff
@@ -178,15 +177,10 @@ impl<D: Debug> Debug for Maximum<D> {
...
-            match &pre.partial_cmp(&next) {
-                Some(Ordering::Less) => {
-                    self.max = Some(next);
-                }
-                Some(Ordering::Equal) => self.max = Some(pre),
-                Some(Ordering::Greater) => self.max = Some(pre),
-                None => Err(FnExecError::accum_error("Data cannot be compared in Maximum Accumulator"))?,
+    fn accum(&mut self, mut next: D) -> FnExecResult<()> {
+        if let Some(pre) = self.max.as_mut() {
+            if pre < &mut next {
```
Comment (longbinlai) [inline research/query_service/ir/runtime/src/process/operator/accum/accumulator.rs:182]: Do not need &mut next, simply &next, and hence can remove `mut` in `mut next: D`

Comment (longbinlai) [reply research/query_service/ir/runtime/src/process/operator/accum/accumulator.rs:182]: You can do (pre as &D) < &next

Comment (BingqingLyu) [reply research/query_service/ir/runtime/src/process/operator/accum/accumulator.rs]: Done

**Pull request merged**